compiler/src/dotty/tools/dotc/parsing/Parsers.scala

```diff
@@ -1339,12 +1330,30 @@ object Parsers {
           WhileDo(cond, body)
         }
       case DO =>
-        atSpan(in.skipToken()) {
+        in.errorOrMigrationWarning(
+          i"""`do <body> while <cond>' is no longer supported,
+             |use `while {<body> ; <cond>} do ()' instead.
```

Comment:
I'm pretty sure we discussed today that this PR shouldn't be mixed up with that possible future syntax. Hence I believe we should suggest `while ({ body; cond }) ()` here instead.